<a href="https://www.kaggle.com/code/somkeneuzowulu/titanic-machine-learning-from-disaster?scriptVersionId=128263694" target="_blank"><img align="left" alt="Kaggle" title="Open in Kaggle" src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>

In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory
import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/titanic/train.csv
/kaggle/input/titanic/test.csv
/kaggle/input/titanic/gender_submission.csv


After downloading the input dataset and viewing their directory output, I copied these respective directories to create DataFrame Objects, ***train_data*** and ***test_data*** in order to view my train.csv and test.csv files. 

For my training Datasets, I noticed that the 'Age' column contains lots of missing data. Therefore, I decided to create a Pandas boolean object, ***bool_series*** 

The **bool_series** object will allow me to access only the passengers of my training Datasets with *complete* details. By doing this, I will train my Machine learning model with a complete Dataset.

In [2]:
train_data = pd.read_csv('/kaggle/input/titanic/train.csv')
bool_series = pd.notnull(train_data['Age'])
train_data = train_data[bool_series]
train_data

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
885          886         0       3   
886          887         0       2   
887          888         1       1   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
885               Rice, Mrs. William (Margaret Norton)  female  39.0      0   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
885      5            382652  29.1250   NaN        Q  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[714 rows x 12 columns]

In [3]:
features = ['Pclass', 'Male', 'Age', 'SibSp', 'Parch', 'Fare']
train_data['Male'] = train_data['Sex'] == 'male'


y = train_data['Survived'].values
X = train_data[features].values

print(X)

[[3 True 22.0 1 0 7.25]
 [1 False 38.0 1 0 71.2833]
 [3 False 26.0 0 0 7.925]
 ...
 [1 False 19.0 0 0 30.0]
 [1 True 26.0 0 0 30.0]
 [3 True 32.0 0 0 7.75]]


A Machine Learning model for the Titanic DataSet will be built using Random Forest.

In [4]:
from sklearn.ensemble import RandomForestClassifier

rf = RandomForestClassifier(n_estimators=100)
rf.fit(X,y)
rf.score(X,y)

0.9859943977591037

For my test Dataset, there are still missing data values in the 'Age column. But I cannot ignore rows that contain missing 'Age' values because it will not allow me to make predictions for such rows.

Therefore, I will fill in those missing data in the 'Age' column by using the **fillna()** function. This function will fill up missing data in a Dataframe with value of the function's argument as shown below:

In [5]:
test_data = pd.read_csv('/kaggle/input/titanic/test.csv')

features = ['Pclass', 'Male', 'Age', 'SibSp', 'Parch', 'Fare']
test_data['Male'] = test_data['Sex'] == 'male'

test_data = test_data.fillna(0)
X_test = test_data[features].values
test_data

PassengerId  Pclass                                          Name  \
0            892       3                              Kelly, Mr. James   
1            893       3              Wilkes, Mrs. James (Ellen Needs)   
2            894       2                     Myles, Mr. Thomas Francis   
3            895       3                              Wirz, Mr. Albert   
4            896       3  Hirvonen, Mrs. Alexander (Helga E Lindqvist)   
..           ...     ...                                           ...   
413         1305       3                            Spector, Mr. Woolf   
414         1306       1                  Oliva y Ocana, Dona. Fermina   
415         1307       3                  Saether, Mr. Simon Sivertsen   
416         1308       3                           Ware, Mr. Frederick   
417         1309       3                      Peter, Master. Michael J   

        Sex   Age  SibSp  Parch              Ticket      Fare Cabin Embarked  \
0      male  34.5      0      0              330911    7.8292     0        Q   
1    female  47.0      1      0              363272    7.0000     0        S   
2      male  62.0      0      0              240276    9.6875     0        Q   
3      male  27.0      0      0              315154    8.6625     0        S   
4    female  22.0      1      1             3101298   12.2875     0        S   
..      ...   ...    ...    ...                 ...       ...   ...      ...   
413    male   0.0      0      0           A.5. 3236    8.0500     0        S   
414  female  39.0      0      0            PC 17758  108.9000  C105        C   
415    male  38.5      0      0  SOTON/O.Q. 3101262    7.2500     0        S   
416    male   0.0      0      0              359309    8.0500     0        S   
417    male   0.0      1      1                2668   22.3583     0        C   

      Male  
0     True  
1    False  
2     True  
3     True  
4    False  
..     ...  
413   True  
414  False  
415   True  
416   True  
417   True  

[418 rows x 12 columns]

The missing data in the Test Dataset was filled with '0'. After this,the prediction of survival for the 418 passengers in the test Dataset is stored and saved as submission_titanic.csv:

In [6]:
y_pred = rf.predict(X_test)
output = pd.DataFrame({'PassengerId': test_data.PassengerId, 'Survived': y_pred})
output.to_csv('submission_titanic.csv', index=False)
print("Submission file created successfully")

Submission file created successfully
